File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.py
class Solution:
  def maxAscendingSum(self, nums: List[int]) -> int:
    ans = s = nums[0]
    for i in range(1, len(nums)):
      if nums[i-1] >= nums[i]:
        ans, s = max(ans, s), 0
      s += nums[i]
    return max(ans, s)
        

File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.ts
function maxAscendingSum(nums: number[]): number {
  let ans = nums[0], sum = nums[0];
  for (let i = 1; i < nums.length; i++) {
    if (nums[i-1] >= nums[i]) {
      [ans, sum] = [Math.max(ans, sum), 0];
    }
    sum += nums[i];
  }
  return Math.max(ans, sum);
};